sycl/include/sycl/ext/intel/esimd/detail/atomic_intrin.hpp

```diff
@@ -16,12 +16,18 @@ namespace __ESIMD_DNS {
 // This function implements atomic update of pre-existing variable in the
 // absense of C++ 20's atomic_ref.
 
+// __atomic_* functions support only integral types. In order to
+// support floating types for certain operations like min/max,
+// 'cmpxchg' operation is applied for result values using
+// 'bridging' variables in integral type.
+template <typename Ty> using CmpxchgTy = __ESIMD_DNS::uint_type_t<sizeof(Ty)>;
+
 template <typename Ty> Ty atomic_load(Ty *ptr) {
 #ifdef _WIN32
   // TODO: Windows will be supported soon
   __ESIMD_UNSUPPORTED_ON_HOST;
 #else
-  return __atomic_load(ptr, __ATOMIC_SEQ_CST);
+  return __atomic_load_n((CmpxchgTy<Ty> *)ptr, __ATOMIC_SEQ_CST);
```

Comment:
`type __atomic_load_n (type *ptr, int memorder)`
so AFAIU `return __atomic_load_n((CmpxchgTy<Ty> *)ptr, __ATOMIC_SEQ_CST);` does type conversion from CmpxchgTy<Ty> and Ty, which is wrong. Do you have tests for this? They should have caught this problem.
```suggestion
  return sycl::bit_cast<Ty>(__atomic_load_n((CmpxchgTy<Ty> *)ptr, __ATOMIC_SEQ_CST));
```

Reply:
I don't have a unit test for this. I'll put TODO comments with `__ESIMD_UNSUPPORTED_ON_HOST`.